frontend/dockerfile/builder/caps.go

```diff
@@ -0,0 +1,33 @@
+package builder
+
+import (
+	"strings"
+
+	"github.com/moby/buildkit/solver/errdefs"
+	"github.com/moby/buildkit/util/grpcerrors"
+	"github.com/moby/buildkit/util/stack"
+	"google.golang.org/grpc/codes"
+)
+
+var enabledCaps = map[string]struct{}{
+	"moby.buildkit.frontend.inputs": {},
+}
+
+func validateCaps(req string) (follow bool, err error) {
+	if req == "" {
+		return
+	}
+	caps := strings.Split(req, ",")
+	for _, c := range caps {
+		parts := strings.SplitN(c, "+", 2)
+		if _, ok := enabledCaps[parts[0]]; !ok {
+			err = stack.Enable(grpcerrors.WrapCode(errdefs.NewUnsupportedFrontendCapError(parts[0]), codes.Unimplemented))
+			if strings.Contains(c, "+follow") {
```

Comment:
What does this `follow` mean?

Reply:
This is for the case where a new cap is added to external frontend but builtin dockerfile frontend does not know about it and the external frontend is loaded through `#syntax`. Then you can use `cap.name+follow` and if dockerfile frontend doesn't know about it but redirects build request through `#syntax` anyway, it will ignore it and let external frontend handle it.

For example frontend inputs cap does not use follow because if builtin dockerfile doesn't support it, it also can't forward the inputs properly. But in most future cases this shouldn't apply.

Renamed to "forward" that I think is a better name for the behavior.